Base PHP-cs-fixer migration to version 2 plus style fix

massimilianobraglia opened the pull request:
Hey there!

As stated here https://github.com/ruflin/Elastica/pull/1540, I'm opening multiple CS fix PRs.

This is the base porting from version 1 to version 2 of the linter. Also, I have applied the CS rules in order to show you how this should look like. 

@ruflin and @p365labs: what's your opinion about this?

ruflin commented:
@massimilianobraglia This is great and it looks like it could still be reviewable as the changes overall are minimal.

ruflin commented on env/elastica/composer.json:
```diff
@@ -13,14 +13,14 @@
-        "sebastian/phpcpd":"~2.0",
-        "squizlabs/php_codesniffer":"~2.5"
+        "phpdocumentor/phpdocumentor": "~2.8",
+        "friendsofphp/php-cs-fixer": "^2.0",
```
Could you have a look at CI? I assume some more updates are needed: https://travis-ci.org/ruflin/Elastica/jobs/467233662

massimilianobraglia replied:
Ok! I'll check it ASAP

massimilianobraglia commented on Makefile:
```diff
@@ -140,7 +140,7 @@ gource:
 ## DOCKER IMAGES
 
-.PHONY: elastica-image
+	.PHONY: elastica-image
```
No, I didn't notice that 🤔
Fixing..

ruflin replied:
still needs a fix ;-)

massimilianobraglia replied:
fixed 😄

massimilianobraglia commented on lib/Elastica/Index/Settings.php:
```diff
@@ -87,23 +88,22 @@ public function get($setting = '')
-                    } else {
-                        return;
-                    }
+        }
```
Yeps! It's such a nice tool (and there are many rules that I have to look deeper 😄)

massimilianobraglia commented on test/Elastica/Aggregation/BucketScriptTest.php:
```diff
@@ -89,20 +90,22 @@ public function testConstructThroughSetters()
      */
     public function testToArrayInvalidBucketsPath()
     {
+        $this->expectException(\Elastica\Exception\InvalidException::class);
```
Yes, it should be done by using the `'php_unit_dedicate_assert' => ['target' => 'newest'],` rule

massimilianobraglia commented on test/Elastica/Base.php:
```diff
@@ -7,7 +8,7 @@
 use Psr\Log\LoggerInterface;
 
-class Base extends \PHPUnit_Framework_TestCase
+class Base extends \PHPUnit\Framework\TestCase
```
Yes. Obvioulsy it should be imported with the `use` statement

massimilianobraglia commented:
Everything touched by this PR is automated by the CS-fixer. A part from the `Makefile` change :/ (probably is my mistake).

About the composer problem, it must be a version conflict for `phpdocumentor/fileset` (I think). Still I didn't have time to check it, what do you think about it @ruflin?

ruflin commented:
yes, seems like it. I wonder if setting it to 2.9 helps?

massimilianobraglia commented:
Unfortunately it does not help. I tried to remove it also and it does not work anyway.

Also, I noticed that the CS-Fixer does not support PHP 7.3 (see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3697)...

ruflin commented:
The good news at least the errors change now. I'm opening https://github.com/ruflin/Elastica/pull/1567 so we can experimenting with updating the other dependencies separately.

For php-cs-fixer: When do you expect it to be compatible with PHP 7.3? Is it only a problem with 7.3 if it runs the linter or always? I'm thinking that on travis we should run linting as a separate step and not on all 4 versions.

p365labs commented:
@massimilianobraglia I think we need to update composer.json here

```env\elastica\composer.json```

it uses         "phpunit/phpunit":"~5.6"

not sure bad I have a bad smell on that version ;)

massimilianobraglia commented:
@ruflin the issue I linked is opened since April 2018. I do not know when the full compatibility will be merged. A comment states that the linter is compatible since we do not use one of the new features of PHP 7.3. I agree about not using the linter in every step but in another one.

massimilianobraglia commented:
I made it! Let's see what the build on Travis says but the problem about composer's hangin' on was also with the phploc package (see https://github.com/ruflin/Elastica/commit/657da466a551ad99830d335d712f114bb606f656#diff-bdfe059178421755f0677c4b7b332938L20)

I just removed it and it worked!

I have also updated PHPUnit version to ^6.0.

massimilianobraglia commented:
https://github.com/lmc-eu/steward/pull/230

What about using this workaround until PHP-CS-fixer solves the issue with PHP 7.3?

ruflin commented:
I assume we don't need a workaround as we just run the linting job with 7.2?

Could you rebase this PR on top of master as @p365labs updated the dependency. I hope now the build will run.

massimilianobraglia commented:
I made it running. The build failed beacuse of PHP-CS-Fixer on PHP 7.3.

Rebasing will be done in a moment 😄

p365labs commented:
@massimilianobraglia well done ! good point amending .travis + Make 👍

p365labs commented on env/elastica/Docker70:
```diff
@@ -34,5 +34,7 @@ ENV PATH=/root/composer/vendor/bin:$PATH
 
 COPY composer.json /root/composer/
 
+RUN composer global require --no-update friendsofphp/php-cs-fixer:^2.0
```
I really don't like it putting it here, but I understand it could be the only solution until php-cs-fixer is released for PHP 7.3

massimilianobraglia commented:
That's the point in fact..!

ruflin commented on env/elastica/composer.json:
```diff
@@ -13,8 +13,6 @@
     ],
     "require": {
         "php": "^7.0",
-        "phpdocumentor/phpdocumentor": "^2.9",
```
Can we please keep this in? I need this to generate the docs :-)

massimilianobraglia replied:
Ok :) Probably I forgot to readd that one.

massimilianobraglia commented:
It still causes the issue. No problem about adding a separate step with the linting :)
Let's see if I can manage it right now

ruflin commented:
@massimilianobraglia LGTM

massimilianobraglia commented:
What should I enter in the CHANGELOG?
Something like
`Introduced new version of PHP-CS-Fixer and new Lint travis step`?

ruflin commented:
@massimilianobraglia Changelog SGTM.

massimilianobraglia commented:
here you are 😄

ruflin commented:
@massimilianobraglia Merged 🎄 Thank you so much for going through all these loops to get this in. 

Moving forward we should make sure it's as easy as possible for contributors to apply the linting to not get stuck on this step in their PR's. 

I'm happy to apply more linting rules to Elastica. The less we need to discuss about how things are formatted and the more we can focus on the code itself the better :-)

ruflin commented:
@massimilianobraglia Yes, please. Best do 1 PR per change so we keep it small.
